src/nbsap/views/eu_strategy.py

```diff
@@ -3,28 +3,61 @@
 from django.contrib import messages
 from django.http import HttpResponse
 from django.utils.translation import ugettext_lazy as _
-
+from nbsap.models import sort_by_code
+from nbsap.views.goals import get_adjacent_targets
 
 from nbsap import models
 from auth import auth_required
 
-from nbsap.forms import EuTargetForm, EuTargetEditForm
+from nbsap.forms import EuTargetForm, EuTargetEditForm, RegionForm
 
 
-def eu_targets(request, pk):
-    current_target = get_object_or_404(models.EuTarget, pk=pk)
-    targets = models.EuTarget.objects.all()
+def get_most_relevant_aichi_targets(target):
+    if not target:
+        return None, None
+    most_relevant_aichi_targets = []
+    for strategy in target.eu_aichi_strategy.all():
+        for aichi_target in strategy.aichi_targets.all():
+            most_relevant_aichi_targets.append(aichi_target)
+    return sort_by_code(most_relevant_aichi_targets)
```

Comment:
This function returns two different types: *tuple* and *list*, and the tuple is not an empty one, it contains two values. Why is that? I think you should return an empty list when there is no `target` available.